packages/swap/src/handlers/getSwapStatus.ts

```diff
@@ -0,0 +1,191 @@
+        issuedBlock_srcChain_channelId: {
+          issuedBlock: Number(issuedBlock),
+          srcChain: srcChain as Chain,
+          channelId: BigInt(channelId),
+        },
+      },
+      include: {
+        swaps: {
+          include: { egress: { include: { broadcast: true } } },
+          orderBy: { id: 'desc' },
+        },
+      },
+    });
+
+    if (!swapDepositChannel) {
+      logger.info(`could not find swap request with id "${id}`);
+      throw ServiceError.notFound();
+    }
+
+    swap = swapDepositChannel.swaps.at(0);
+  } else if (swapIdRegex.safeParse(id).success) {
+    swap = await prisma.swap.findUnique({
+      where: { nativeId: BigInt(id) },
+      include: { egress: { include: { broadcast: true } } },
+    });
+  } else if (txHashRegex.safeParse(id).success) {
+    swap = await prisma.swap.findFirst({
+      where: { txHash: id },
+      include: { egress: { include: { broadcast: true } } },
+      // just get the last one for now
+      orderBy: { createdAt: 'desc' },
+    });
+  }
+
+  ServiceError.assert(
+    swapDepositChannel || swap,
+    'notFound',
+    'resource not found',
+  );
+
+  let state: z.infer<typeof swapState>;
+
+  if (swap?.egress?.broadcast?.succeededAt) {
+    assert(swap.swapExecutedAt, 'swapExecutedAt should not be null');
+    state = swapState.enum.COMPLETE;
+  } else if (swap?.egress?.broadcast?.abortedAt) {
+    assert(swap.swapExecutedAt, 'swapExecutedAt should not be null');
+    state = swapState.enum.BROADCAST_ABORTED;
+  } else if (swap?.egress?.broadcast) {
+    assert(swap.swapExecutedAt, 'swapExecutedAt should not be null');
+    state = swapState.enum.BROADCAST_REQUESTED;
+  } else if (swap?.egress) {
+    assert(swap.swapExecutedAt, 'swapExecutedAt should not be null');
+    state = swapState.enum.EGRESS_SCHEDULED;
+  } else if (swap?.swapExecutedAt) {
+    state = swapState.enum.SWAP_EXECUTED;
+  } else if (swap?.depositReceivedAt) {
+    state = swapState.enum.DEPOSIT_RECEIVED;
+  } else {
+    state = swapState.enum.AWAITING_DEPOSIT;
+  }
+
+  const srcAsset = readField(swap, swapDepositChannel, 'srcAsset');
+  const destAsset = readField(swap, swapDepositChannel, 'destAsset');
+  assert(srcAsset && destAsset, 'srcAsset and destAsset should be defined');
+
+  let pendingDeposit;
+
+  if (
+    state === swapState.enum.AWAITING_DEPOSIT &&
+    swapDepositChannel?.depositAddress
+  ) {
+    pendingDeposit = await getPendingDeposit(
+      assetChains[srcAsset],
+      swapDepositChannel.depositAddress,
+    );
+  }
+
+  let estimatedDepositChannelExpiryTime: Date | null = null;
+  if (swapDepositChannel) {
+    const chainInfo = await prisma.chainTracking.findFirst({
+      where: {
+        chain: swapDepositChannel.srcChain,
+      },
+    });
+    estimatedDepositChannelExpiryTime = calculateExpiryTime({
+      chain: swapDepositChannel.srcChain,
+      startBlock: chainInfo?.height,
+      expiryBlock: swapDepositChannel.srcChainExpiryBlock,
+    });
+  }
```

Comment:
This is changed recently in the original `routes/swap.ts` file, can you please update it here too?